src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -31,11 +30,14 @@ public ParsedReport parse(Report report) {
 
             List<LetterPrintStatus> statuses =
                 stream(parser.spliterator(), false)
-                    .map(row -> toPrintStatus(row))
-                    .filter(Objects::nonNull)
+                    .map(csvRecord -> toPrintStatus(csvRecord))
                     .collect(toList());
 
-            return new ParsedReport(report.path, statuses);
+            return new ParsedReport(
+                report.path,
+                statuses.stream().filter(s -> s != null).collect(toList()),
+                statuses.stream().allMatch(s -> s != null)
```

Comment:
can be replaced with `anyMatch` and short circuit the whole process. will work for empty too.

Reply:
can we also use object references in both cases instead?

Reply:
`allMatch` will stop on first null as well.
I prefer lambdas in this case

Reply:
ah, skipped that sentence in docblock

Reply:
fair enough. my two cents would be reusability and since we are keen on making code as humanly readable as possible would only make super obvious in this case `filter objects that are non null`. but as internet says - personal preference overtakes this one

and anyway - non blocker. i was more interested in stream usage